Apps/CMakeLists.txt

```diff
@@ -21,35 +21,64 @@ set(EXECUTABLE_OUTPUT_PATH ${CMAKE_BINARY_DIR}/bin)
 
 if(EXISTS ${PROJECT_SOURCE_DIR}/IFEM-Poisson)
   add_subdirectory(IFEM-Poisson)
+elseif(EXISTS ${PROJECT_SOURCE_DIR}/Poisson)
+  add_subdirectory(Poisson)
 endif()
+
 if(EXISTS ${PROJECT_SOURCE_DIR}/IFEM-AdvectionDiffusion)
-        add_subdirectory(IFEM-AdvectionDiffusion)
+  add_subdirectory(IFEM-AdvectionDiffusion)
+elseif(EXISTS ${PROJECT_SOURCE_DIR}/AdvectionDiffusion)
+  add_subdirectory(AdvectionDiffusion)
 endif()
+
 if(EXISTS ${PROJECT_SOURCE_DIR}/IFEM-Darcy)
-        add_subdirectory(IFEM-Darcy)
+  add_subdirectory(IFEM-Darcy)
+elseif(EXISTS ${PROJECT_SOURCE_DIR}/Darcy)
+  add_subdirectory(Darcy)
 endif()
 
 if(EXISTS ${PROJECT_SOURCE_DIR}/IFEM-Elasticity)
-        add_subdirectory(IFEM-Elasticity)
-        add_subdirectory(IFEM-Elasticity/Beam)
-        add_subdirectory(IFEM-Elasticity/Linear)
-        if(EXISTS ${PROJECT_SOURCE_DIR}/IFEM-Elasticity/IFEM-BeamEx)
-                add_subdirectory(IFEM-Elasticity/IFEM-BeamEx)
-                add_subdirectory(IFEM-Elasticity/IFEM-BeamEx/BeamSim)
+  add_subdirectory(IFEM-Elasticity)
+  add_subdirectory(IFEM-Elasticity/Beam)
+  add_subdirectory(IFEM-Elasticity/Linear)
+  if(EXISTS ${PROJECT_SOURCE_DIR}/IFEM-Elasticity/IFEM-BeamEx)
+    add_subdirectory(IFEM-Elasticity/IFEM-BeamEx)
+    add_subdirectory(IFEM-Elasticity/IFEM-BeamEx/BeamSim)
   endif()
   if(EXISTS ${PROJECT_SOURCE_DIR}/IFEM-Elasticity/IFEM-FiniteDeformation)
-          add_subdirectory(IFEM-Elasticity/IFEM-FiniteDeformation)
-          add_subdirectory(IFEM-Elasticity/IFEM-FiniteDeformation/Nonlinear)
+    add_subdirectory(IFEM-Elasticity/IFEM-FiniteDeformation)
+    add_subdirectory(IFEM-Elasticity/IFEM-FiniteDeformation/Nonlinear)
   endif()
   if(EXISTS ${PROJECT_SOURCE_DIR}/IFEM-ThermoElasticity)
-          add_subdirectory(IFEM-ThermoElasticity)
+    add_subdirectory(IFEM-ThermoElasticity)
   endif()
   if(EXISTS ${PROJECT_SOURCE_DIR}/IFEM-PoroElasticity)
     add_subdirectory(IFEM-PoroElasticity)
   endif()
   if(EXISTS ${PROJECT_SOURCE_DIR}/IFEM-OpenFrac)
     add_subdirectory(IFEM-OpenFrac)
   endif()
+elseif(EXISTS ${PROJECT_SOURCE_DIR}/Elasticity)
```

Comment:
this can be deduplicated by setting some var elasticity_root var first.

Reply:
Tried something like that in the last fixup commit. Not sure if this is the best way but think it works?